src/pandapipes/component_models/heat_consumer_component.py

```diff
@@ -0,0 +1,231 @@
+        hs_pit[:, VINIT] = (net[cls.table_name()].controlled_mdot_kg_per_s.values /
+                            (hs_pit[:, AREA] * hs_pit[:, RHO]))
+        hs_pit[:, ALPHA] = 0
+        hs_pit[:, QEXT] = net[cls.table_name()].qext_w.values
+        hs_pit[:, TEXT] = 293.15
+        return hs_pit
+
+    @classmethod
+    def create_component_array(cls, net, component_pits):
+        """
+        Function which creates an internal array of the component in analogy to the pit, but with
+        component specific entries, that are not needed in the pit.
+
+        :param net: The pandapipes network
+        :type net: pandapipesNet
+        :param component_pits: dictionary of component specific arrays
+        :type component_pits: dict
+        :return:
+        :rtype:
+        """
+        tbl = net[cls.table_name()]
+        consumer_array = np.zeros(shape=(len(tbl), cls.internal_cols), dtype=np.float64)
+        consumer_array[:, cls.MASS] = tbl.controlled_mdot_kg_per_s.values
+        consumer_array[:, cls.QEXT] = tbl.qext_w.values
+        consumer_array[:, cls.DELTAT] = tbl.deltat_k.values
+        consumer_array[:, cls.TRETURN] = tbl.treturn_k.values
+        mf = ~np.isnan(consumer_array[:, cls.MASS])
+        qe = ~np.isnan(consumer_array[:, cls.QEXT])
+        dt = ~np.isnan(consumer_array[:, cls.DELTAT])
+        tr = ~np.isnan(consumer_array[:, cls.TRETURN])
+        consumer_array[:, cls.MODE] = np.sum([mf, qe, dt, tr], axis=0)
+        component_pits[cls.table_name()] = consumer_array
+
+    @classmethod
+    def adaption_after_derivatives_hydraulic(cls, net, branch_pit, node_pit, idx_lookups, options):
+        """
+        Perform adaptions to the branch pit after the derivatives have been calculated globally.
+
+        :param net: The pandapipes network containing all relevant info
+        :type net: pandapipesNet
+        :param branch_pit: The branch internal array
+        :type branch_pit: np.ndarray
+        :param node_pit: The node internal array
+        :type node_pit: np.ndarray
+        :param idx_lookups: Lookup for the relevant indices in the pit
+        :type idx_lookups: dict
+        :param options: Options for the pipeflow
+        :type options: dict
+        :return: No Output.
+        :rtype: None
+        """
+        # set all pressure derivatives to 0 and velocity to 1; load vector must be 0, as no change
+        # of velocity is allowed during the pipeflow iteration
+        f, t = idx_lookups[cls.table_name()]
+        fc_branch_pit = branch_pit[f:t, :]
+        fc_array = get_component_array(net, cls.table_name())
+        # TODO: this is more precise, but slower:
+        #       np.isin(fc_array[:, cls.MODE], [cls.MF_QE, cls.MF_DT, cls.MF_TR])
+        mdot_controlled = ~np.isnan(fc_array[:, cls.MASS])
+        fc_branch_pit[mdot_controlled, JAC_DERIV_DP] = 0
+        fc_branch_pit[mdot_controlled, JAC_DERIV_DP1] = 0
+        fc_branch_pit[mdot_controlled, JAC_DERIV_DV] = 1
```

Comment:
change to JAC_DERIV_DM

Reply:
We should check which PR to merge first... don't know which order is better :-D

Reply:
hehe, yes sure. I guess, it doesn't make a big difference :D